### ISSUE TITLE
COAM\ Earned Credits issuance request actioned by CAS

### ISSUE THREAD

**Issue by NicoleGovvy** [bcgov/cas-coam#65]
#### Description:

CAS director has clicked the approve or decline button for the EC credit issuance request, the system needs to update.

#### Acceptance Criteria:

For an approval:
- [ ] The status displayed on the ['Track Status of Issuance' page](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=21483-5360&t=hGt2AecQ4gUYTU8G-4) is now 'Approved, to be issued in BCCR'
- [ ] If the API allows credits to be automatically credited to the BCCR account, the status will update to 'Issued in BCCR' automatically
- [ ] If the API does not allow for this, clicking the 'Approve' button will send an email to ghgregulator@gov.bc.ca to notify staff to manually issue credits to the specified BCCR holding account

If credits need to be issued manually in BCCR, the status wording would be updated to 'Issuance approved', and language about OR staff would need to be able to manually change the status of the issuance to 'Issued in BCCR'

For a decline:

- [ ] The status displayed on the 'Track status of issuance' page would be Issuance declined
- [ ] Credits are not issued to the BCCR account

#### Development Checklist:

- [ ] Checklist item
- [ ] Checklist item
- [ ] Checklist item
- [ ] Meets the DOD

#### Definition of Ready (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

#### Definition of Done (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Notes:

- This ticket depends on whether or not the API can push credits to a holding account. If not, the wording on the statuses will need to be reviewed, or CAS staff will need the ability to manually change the status to 'Issued in BCCR' once they've done that. KK believes account holders are issued an email in BCCR when credits issued so that could be the notification piece.
- This ticket depends on how the business area needs to see and interact with issuance requests after they're completed - can they edit them? How do they view a list of closed issuance requests? What happens when there's more than one transaction involving re-dos or lower amounts?

**Comment by NicoleGovvy** [comment]
Outstanding questions:

This ticket depends on whether or not the API can push credits to a holding account. If not, the wording on the statuses will need to be reviewed, or CAS staff will need the ability to manually change the status to 'Issued in BCCR' once they've done that. KK believes account holders are issued an email in BCCR when credits issued so that could be the notification piece.

This ticket depends on how the business area needs to see and interact with issuance requests after they're completed - can they edit them? How do they view a list of closed issuance requests? What happens when there's more than one transaction involving re-dos or lower amounts?

**Comment by fviduya** [comment]
I have tested this in the DEV environment, [here is the test document for reference](https://buttoninc-my.sharepoint.com/:w:/g/personal/francesca_button_is/EeG5rUxf8MxGkpJa4k0DDvIB-PEdchaf7gI1HykqXGvTyw?e=45Ay9S).

I can verify that the Track Status of Issuance page:
- [x] Displays in the DEV environment
- [x] Is identical to the [wireframe](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=21296-1319&t=xjYJ0gAr6kKncV1V-0)
   - [ ] Breadcrumbs show the operation name
   - [x] Operation name is visible at the top left of the page
   - [x] `Track Status of Issuance` heading is visible
   - [x] A note is visible below the heading, currently it shows pending with a note: "Your request has been submitted successfully. Once your request is approved, the earned credits will be issued to your holding account as identified below in the B.C. Carbon Registry (BCCR)."
   - [x] `Status of Issuance` subheading is visible, below that:
      - [x] `Earned Credits`, `Status of Issuance`, and `BCCR Trading Name` are visible

Per @nataliepiguliak, the Approve/Decline button functionality will be implemented in #166, it is currently not interactive and therefore cannot change the status of the EC issuance request. The verification for this ticket is to verify that the UI is similar to the wireframe. The only thing that does not match the wireframe is the operation name not visible in the breadcrumbs. Moving to Pending PO for review of test, please let me know if I missed anything else. Thank you!

**Comment by vesselak** [comment]
Reviewed testing documentation and accepting ticket
